packages/wrangler/src/deployment-bundle/resolve-entry.ts

```diff
@@ -12,31 +13,33 @@ export function resolveEntryWithScript(script: string): {
 
 export function resolveEntryWithMain(
 	main: string,
-	configPath?: string
+	config: Config
 ): {
 	absolutePath: string;
 	relativePath: string;
 	projectRoot: string;
 } {
-	const projectRoot = path.resolve(path.dirname(configPath ?? "."));
-	const file = path.resolve(projectRoot, main);
-	const relativePath = path.relative(projectRoot, file) || ".";
-	return { absolutePath: file, relativePath, projectRoot };
+	const projectRoot = path.resolve(path.dirname(config.userConfigPath ?? "."));
```

Comment:
could we add a comment here explaining why the `projectRoot` and `entryRoot` can potentially diverge into 2 diff paths?

Reply:
also I wonder if we won;t need to make this kind of distinction in other code paths as well, in which case we'd prolly want to extract these two into a fn that returns them? Still going through the code, so might be over engineering...I'll defer to your judgement. Just wanted to call it out

Reply:
This is the main place I believe where it matters. The `getEntry()` is effectively the helper that covers this stuff.